app/src/main/java/org/kontalk/util/MessageUtils.java

```diff
@@ -708,17 +735,29 @@ public static String toString(byte[] text) {
         return text;
     }
 
-    public static void convertSmileys(Editable input) {
+    public static boolean convertSmileys(Editable input) {
+        int i = 0;
         for (String key : sEmojiConverterMap.keySet()) {
-            replaceEditable(input, key, sEmojiConverterMap.get(key));
+            if (replaceEditable(input, key, sEmojiConverterMap.get(key))){
+                i++;
+            }
+        }
+        if (i>0){
+            return true;
         }
+        else return false;
     }
 
-    private static void replaceEditable(Editable text, String in, String out) {
-        int position = text.toString().indexOf(in);
-        if (position >= 0) {
+    private static boolean replaceEditable(Editable text, String in, String out) {
+        int i = 0;
+        for (int position = text.toString().indexOf(in); position >= 0; position = text.toString().indexOf(in)){
             text.replace(position, position + in.length(), out);
+            i++;
```

Comment:
same as above, can be replaced with a bool. (still don't see why you want it to be a bool)